plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/launcher/PipelineLaunchConfiguration.java

```diff
@@ -146,30 +171,23 @@ private void setValuesFromLaunchConfiguration(ILaunchConfiguration configuration
 
     setUserOptionsName(configuration.getAttribute(
         PipelineConfigurationAttr.USER_OPTIONS_NAME.toString(), getUserOptionsName()));
-
-    setEclipseProjectName(configuration);
   }
 
   /**
    * Stores the Dataflow Pipeline-specific options of this LaunchConfiguration inside the provided
    * {@link ILaunchConfigurationWorkingCopy}.
    */
   public void toLaunchConfiguration(ILaunchConfigurationWorkingCopy configuration) {
-    configuration.setAttribute(
-        PipelineConfigurationAttr.RUNNER_ARGUMENT.toString(), runner.getRunnerName());
+    if (runner != null) {
+      configuration.setAttribute(PipelineConfigurationAttr.RUNNER_ARGUMENT.toString(),
+          runner.getRunnerName());
```

Comment:
Note to myself: save a runner in LC after selecting a project, save, close, unset a project, save, close, and see if PLC is constructed without a runner.